apps/transport/lib/jobs/resource_history_job.ex

```diff
@@ -0,0 +1,191 @@
+defmodule Transport.Jobs.ResourceHistoryDispatcherJob do
+  @moduledoc """
+  Job in charge of dispatching multiple `ResourceHistoryJob`
+  """
+  use Oban.Worker, unique: [period: 60 * 60 * 5], tags: ["history"], max_attempts: 5
+  require Logger
+  import Ecto.Query
+  alias DB.{Repo, Resource}
+
+  @impl Oban.Worker
+  def perform(_job) do
+    Transport.S3.create_bucket_if_needed!(:history)
+
+    duplicates =
+      Resource
+      |> where([r], not is_nil(r.datagouv_id))
+      |> group_by([r], r.datagouv_id)
+      |> having([r], count(r.datagouv_id) > 1)
+      |> select([r], r.datagouv_id)
+      |> Repo.all()
+
+    datagouv_ids =
+      Resource
+      |> where([r], not is_nil(r.url) and not is_nil(r.title) and not is_nil(r.datagouv_id))
+      |> where([r], r.format == "GTFS" or r.format == "NeTEx")
+      |> where([r], r.datagouv_id not in ^duplicates)
+      |> where([r], not r.is_community_resource)
+      |> select([r], r.datagouv_id)
+      |> Repo.all()
+
+    Logger.debug("Dispatching #{Enum.count(datagouv_ids)} ResourceHistoryJob jobs")
+
+    datagouv_ids
+    |> Enum.map(fn datagouv_id ->
+      %{datagouv_id: datagouv_id} |> Transport.Jobs.ResourceHistoryJob.new()
+    end)
+    |> Oban.insert_all()
+
+    :ok
+  end
+end
+
+defmodule Transport.Jobs.ResourceHistoryJob do
+  @moduledoc """
+  Job historicising a single resource
+  """
+  use Oban.Worker, unique: [period: 60 * 60 * 5], tags: ["history"], max_attempts: 5
+  require Logger
+  import Ecto.Query
+  alias DB.{Repo, Resource}
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"datagouv_id" => datagouv_id}}) do
+    Logger.info("Running ResourceHistoryJob for #{datagouv_id}")
+    resource = Resource |> where([r], r.datagouv_id == ^datagouv_id) |> Repo.one!()
+
+    resource |> download_resource() |> process_download(resource) |> remove_file!()
```

Comment:
Le `remove_file!` gagnerait à être mis dans un block `after` qui sert typiquement à cleaner les resources:

https://elixir-lang.org/getting-started/try-catch-and-rescue.html#after

Pas évident en faisant un pipe là puisque tu as le nom du truc à passer derrière, mais on pourrait simplement refactorer en faisant: création du nom d'abord, puis try/after.

L'idée étant d'éviter de saturer le disque (on a de la place et ce sont des containers temporaires, mais autant assurer le coup là dessus, ça peut vite dégénérer dans le futur).